File: src/types/prompt.ts
export interface Prompt {
  id: string;
  name: string;
  content: string;
  category?: string;
  isSystem: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface PromptInput {
  name: string;
  content: string;
  category?: string;
}

File: src-tauri/src/models/prompt.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Prompt {
    pub id: String,
    pub name: String,
    pub content: String,
    pub category: String,
    pub is_system: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PromptInput {
    pub name: String,
    pub content: String,
    pub category: Option<String>,
}
